src/Diagnostics.DataProviders/MdmClient/MdmClient.cs

```diff
@@ -0,0 +1,451 @@
+                }
+            }
+
+            throw new MetricsClientException($"Exhausted {numAttempts} attempts.", lastException);
+        }
+
+        private static async Task<Tuple<string, HttpResponseMessage>> GetResponseNoRetry(
+                    Uri url,
+                    HttpMethod method,
+                    HttpClient client,
+                    object httpContent,
+                    string clientId,
+                    string serializedContent,
+                    Guid? traceId)
+        {
+            traceId = traceId ?? Guid.NewGuid();
+
+            var request = new HttpRequestMessage(method, url);
+            var sourceId = Environment.MachineName;
+            AddStandardHeadersToMessage(request, traceId.Value, sourceId);
+            if (!string.IsNullOrWhiteSpace(clientId))
+            {
+                request.Headers.Add("ClientId", clientId);
+            }
+
+            if (httpContent != null && serializedContent == null)
+            {
+                serializedContent = JsonConvert.SerializeObject(httpContent);
+            }
+
+            if (serializedContent != null)
+            {
+                request.Content = new StringContent(serializedContent, Encoding.UTF8, "application/json");
+            }
+
+            string responseString = null;
+            var requestLatency = Stopwatch.StartNew();
+            var stage = "SendRequest";
+            var handlingServer = "Unknown";
+            HttpResponseMessage response = null;
+            try
+            {
+                response = await client.SendAsync(request).ConfigureAwait(false);
+
+                stage = "ReadResponse";
+
+                IEnumerable<string> handlingServerValues;
+                response.Headers.TryGetValues("__HandlingServerId__", out handlingServerValues);
+                if (handlingServerValues != null)
+                {
+                    handlingServer = handlingServerValues.First();
+                }
+
+                requestLatency.Restart();
+
+                if (response.Content.Headers.ContentType?.MediaType != null
+                    && response.Content.Headers.ContentType.MediaType.Equals("application/octet-stream", StringComparison.OrdinalIgnoreCase))
+                {
+                    responseString = "application/octet-stream";
+                }
+                else
+                {
+                    responseString = await response.Content.ReadAsStringAsync().ConfigureAwait(false);
+                }
+
+                stage = "ValidateStatus";
+                response.EnsureSuccessStatusCode();
+
+                return Tuple.Create(responseString, response);
+            }
+            catch (Exception e)
+            {
+                var message = $"Failed to get a response from the server. TraceId:{traceId.Value.ToString("B")}, Url:{request.RequestUri}, HandlingServer:{handlingServer} Stage:{stage}, "
```

Comment:
Every time a data provider makes an outbound call and fails we should log the request URL so that we can look at the logs and repro it. So log this message in the finally clause and generalize the message for both success and failures. 

You don't have to log the response string. Unless you know the response string has something very useful for failures.